Implements ECS commands for cloud-compose

uncompiled opened the pull request:
This PR implements cloud-compose commands for ECS clusters.

It supports the standard `up` and `down` commands and implements two new commands:
- `health` - checks whether the ECS cluster is healthy, including the services running on the cluster, whether the number of ECS container instances matches the auto-scaling group desired count, and whether the load balancers for a given service are healthy.
- `upgrade` - this kicks off the upgrade workflow.  State transitions occur when the cluster is healthy, so the upgrade process consists of terminating each of the original ECS container instances in the cluster and allowing the auto-scaling group to replace the instance to match the desired count.  When the cluster is healthy again, the workflow moves to the next instance until all instances have been upgraded.

uncompiled committed:
First pass at cloud-compose ECS plugin

uncompiled committed:
Add ec2 client to describe instances for health check

uncompiled committed:
Remove pretty printer

This was used for printing out the API responses,
but is no longer needed.

jasonbartz commented on cloudcompose/ecs/controller.py:
```diff
@@ -0,0 +1,299 @@
+
+    def _get_ecs_services(self):
+        cluster_services = self._ecs_list_services(cluster=self.name)
+        try:
```
I feel like this is syntactically confusing.  I think you could achieve the same goal by doing:

```python
cluster_services.get("serviceArns")
if cluster_services is not None:
    describe_services = self._ecs_describe_services(cluster=self.name,
                                                 services=cluster_services['serviceArns'])
    return describe_services['services']

raise CloudComposeException("Services could not be retrieved for {}".format(self.name))
```

jasonbartz reviewed (commented):
Good PR, things seem to work well.

I tested this on one of my clusters and incidentally EC2 brought up a bad node.  The ASG caught it and retried and the upgrade process waited and kept going after it came up, so that's excellent.

My feedback is more interface-based.  While a new node was coming up the upgrade script was still spitting out:

```
rhein (i-099e322f7b7d3651e): shutting-down
rhein (i-099e322f7b7d3651e): shutting-down
rhein (i-099e322f7b7d3651e): shutting-down
rhein (i-099e322f7b7d3651e): shutting-down
rhein (i-099e322f7b7d3651e): shutting-down
rhein (i-099e322f7b7d3651e): shutting-down
```

I think it would be more user-friendly to switch from a `shutting-down` message to a starting up message while it's waiting for the new node to get healthy.

uncompiled committed:
Simplify logic when getting services

uncompiled committed:
Update status message to say "replacing" instance

Instead of saying "shutting-down" when printing the status,
tell the user that the instance is being replaced.
